rmw_fastrtps_shared_cpp/src/custom_publisher_info.cpp

```diff
@@ -138,6 +161,20 @@ void PubListener::set_on_new_event_callback(
           incompatible_qos_status_.total_count_change = 0;
         }
         break;
+      case RMW_EVENT_PUBLICATION_MATCHED:
+        {
+          // Not call callback since recorded matched event happened before
```

Comment:
I think this is correct and expected behavior that application does not want to get the matched or unmatched event happened before.

Reply:
Sorry, I don't think so.

I think we might lose a event that users care about.
The case as below,

```
a subscriber(A) is running,
then run a publisher:
   the DDS publisher created.      // the publisher might get the subscriber matched status immediately which happens in the DDS
   set_event_callback (callback)   // callback can't be triggered for the connected subscriber(A).
                                   // if there is no other subscribers comming any more.
   executor.spin()
```

Reply:
Yes. For this case, event may be lost.   

Now except matched/unmatched event, other events all deal with previous event while registering callback. Not sure whether original design may have taken this situation into consideration.   
@fujitatomoya What do you think ?

Reply:
So after all, event that happened before also notifies the user application.

i was not sure about this but thinking about the use case for the application, probably this is useful especially https://github.com/ros2/rmw/issues/330#issue-1371797574 it makes sense to receive the matched/unmatched event that happened before.